components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/gateway/v1/impl/LocalEntryApiServiceImpl.java

```diff
@@ -39,8 +46,19 @@ public Response localEntryGet(String apiName, String label, String apiId, Messag
         JSONArray localEntryArray = new JSONArray();
         if (gatewayAPIDTO != null) {
             if (gatewayAPIDTO.getLocalEntriesToBeAdd() != null) {
+                LocalEntryServiceProxy localEntryServiceProxy = new
+                        LocalEntryServiceProxy(gatewayAPIDTO.getTenantDomain());
                 for (GatewayContentDTO localEntry : gatewayAPIDTO.getLocalEntriesToBeAdd()) {
-                    localEntryArray.put(localEntry.getContent());
+                    try {
+                        if (localEntryServiceProxy.isEntryExists(localEntry.getName())) {
+                            localEntryArray.put(localEntryServiceProxy.getEntry(localEntry.getName()));
+                        } else {
+                            log.error(localEntry.getName() + " was not deployed in the gateway");
```

Comment:
better to have this in the response

Reply:
https://github.com/wso2/carbon-apimgt/pull/8784/commits/2bef8ac27b023cef0271216e1b2942bc739efacc